File: src/main/java/com/claudio_personagens/bookingcharactersapi/repositories/UserRepository.java
package com.claudio_personagens.bookingcharactersapi.repositories;

import com.claudio_personagens.bookingcharactersapi.base.AbstractRepository;
import com.claudio_personagens.bookingcharactersapi.domain.User;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;

public interface UserRepository extends AbstractRepository<User, String>, JpaSpecificationExecutor<User> {
}

File: src/main/java/com/claudio_personagens/bookingcharactersapi/repositories/UserRepository.kt
package com.claudio_personagens.bookingcharactersapi.repositories

import com.claudio_personagens.bookingcharactersapi.base.BaseRepository
import com.claudio_personagens.bookingcharactersapi.security.domain.User
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface UserRepository : BaseRepository<User, UUID> {
    fun findByEmail(email: String): User?
}